### ISSUE TITLE
Improve rule configuration

### ISSUE THREAD

**Issue by jsenko** [Apicurio/apicurio-registry#360]
Rule configuration should not be a single string - the options that rules require should be explicitly defined, multi-valued and possibly type-checked.

See also #359

**Comment by jsenko** [comment]
Consider replacing it by using metadata feature for rules? #361

**Comment by EricWittmann** [comment]
I agree with this.  The problem is how to define rule-specific configuration (options will be different depending on the rule) in the REST API in a sensible way.  It's not impossible certainly, I just punted on doing it for the first go-round.

Perhaps we can define a config data type for each rule type and then combine them using "anyOf".
